app/src/main/java/io/apicurio/registry/ImportLifecycleBean.java

```diff
@@ -0,0 +1,61 @@
+package io.apicurio.registry;
+
+import io.apicurio.registry.storage.RegistryStorage;
+import io.apicurio.registry.storage.impexp.EntityInputStream;
+import io.apicurio.registry.types.Current;
+import io.apicurio.registry.utils.impexp.Entity;
+import io.apicurio.registry.utils.impexp.EntityReader;
+import io.quarkus.runtime.StartupEvent;
+import org.slf4j.Logger;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.zip.ZipInputStream;
+
+@ApplicationScoped
+public class ImportLifecycleBean {
+
+    @Inject
+    Logger log;
+
+    @Inject
+    @Current
+    RegistryStorage storage;
+
+    void onStart(@Observes StartupEvent ev) {
+        final String registryImportZipPath = java.lang.System.getenv("REGISTRY_IMPORT");
```

Comment:
I would change this for a config property like "registry.import" , quarkus translates that to the env variable "REGISTRY_IMPORT" and that will help for implementing a test

Reply:
+1

Reply:
Yep - and it would be good to inject that property via standard config injection.

Reply:
I fully agreed and I've introduced it